tests/data/sentinel2/data.py

```diff
@@ -144,7 +147,75 @@
 }
 
 
-def create_file(path: str, dtype: str, num_channels: int) -> None:
+# Minimal xml content for the gdal Sentinel2 driver to be able to read
+# footprint from tags: `datasource.tags()['FOOTPRINT']`
+# Leave placeholder for coords to be filled with actual valid footprint
+xml_template = """<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<n1:Level-1C_User_Product
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+    xsi:schemaLocation="https://psd-14.sentinel2.eo.esa.int/PSD/User_Product_Level-1C.xsd">
+    <n1:General_Info>
+        <Product_Info>
+            <Query_Options completeSingleTile="true">
+                <PRODUCT_FORMAT>SAFE_COMPACT</PRODUCT_FORMAT>
+            </Query_Options>
+            <Product_Organisation></Product_Organisation>
+        </Product_Info>
+    </n1:General_Info>
+    <n1:Geometric_Info>
+        <Product_Footprint>
+            <Product_Footprint>
+                <Global_Footprint>
+                    <EXT_POS_LIST>{coords}</EXT_POS_LIST>
+                </Global_Footprint>
+            </Product_Footprint>
+        </Product_Footprint>
+    </n1:Geometric_Info>
+</n1:Level-1C_User_Product>
+"""
+
+
+def get_product_root(raster_path: str) -> str:
+    return raster_path.split('GRANULE')[0]
+
+
+def create_metadata_file(raster_path: str) -> None:
+    product_root = get_product_root(raster_path)
+    metadata_path = os.path.join(product_root, 'MTD_MSIL1C.xml')
+
+    # Calculate the actual valid footprint based on pixel values
+    # by specifying nodata value. This will be stored in the metadata file.
+    with rasterio.open(raster_path) as src:
+        source_crs = src.crs
+        valid_data_footprint = shapely.box(*src.bounds)
+        # TODO: Swap to the following when get_valid_footprint_from_datasource exists
+        # with WarpedVRT(src, nodata=0) as vrt:
+        #     valid_data_footprint = shapely.convex_hull(
+        #         get_valid_footprint_from_datasource(vrt)
+        #     )
```

Comment:
Doing it this way we don't need to hard-code the footprint and keep it up-to-date with the actual pixel values.